mmdet/models/detectors/detection_transformer.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import warnings
+
+import torch
+import torch.nn.functional as F
+from mmcv.cnn import Conv2d, xavier_init
+from torch import nn
+
+from mmdet.core import bbox2result
+from ..builder import DETECTORS, build_backbone, build_head, build_neck
+from ..utils.positional_encoding import SinePositionalEncoding
+from ..utils.transformer import DetrTransformerDecoder, DetrTransformerEncoder
+from .base import BaseDetector
+
+
+@DETECTORS.register_module()
+class TransformerDetector(BaseDetector):
```

Comment:
We  can just define the interface in this class and move all the detail implementation to a specific detector